examples/basics/visuals/line_prototype.py

```diff
@@ -0,0 +1,147 @@
+from vispy import app, gloo, visuals
+from vispy.visuals.components import Clipper, Alpha, ColorFilter
+
+        
+
+class LineVisual(visuals.Visual):
+    def __init__(self, pos=None, color=(1, 1, 1, 1)):
+        vcode = """
+        attribute vec2 a_pos;
+        
+        void main() {
+            gl_Position = $transform(vec4(a_pos, 0, 1)); 
+            gl_PointSize = 10;
+        }
+        """
+        
+        fcode = """
+        void main() {
+            gl_FragColor = $color;
+        }
+        """
+        
+        visuals.Visual.__init__(self, vcode=vcode, fcode=fcode)
+        
+        self.pos_buf = gloo.VertexBuffer()
+        self.shared_program['a_pos'] = self.pos_buf
+        self.shared_program.frag['color'] = color
+        self._need_upload = False
+        
+        self._draw_mode = 'line_strip'
+        self.set_gl_state('translucent', depth_test=False)
+        
+        if pos is not None:
+            self.set_data(pos)
+            
+    def set_data(self, pos):
+        self._pos = pos
+        self._need_upload = True
+        
+    def _prepare_draw(self, view=None):
+        if self._need_upload:
+            self.pos_buf.set_data(self._pos)
+            self._need_upload = False
+    
+    @staticmethod
+    def _prepare_transforms(view):
+        tr = view.transforms.get_full_transform()
+        view.view_program.vert['transform'] = tr
+    
+
+class PointVisual(LineVisual):
+    def __init__(self, pos=None, color=(1, 1, 1, 1)):
+        LineVisual.__init__(self, pos, color)
+        self._draw_mode = 'points'
+    
+
+class PlotLineVisual(visuals.CompoundVisual):
+    def __init__(self, pos=None, line_color=(1, 1, 1, 1), point_color=(1, 1, 1, 1)):
+        self._line = LineVisual(pos, color=line_color)
+        self._point = PointVisual(pos, color=point_color)
+        visuals.CompoundVisual.__init__(self, [self._line, self._point])
```

Comment:
So a `CompoundVisual` allows creating a new visual class by combining other existing `Visual` classes. And adding this additional class allows us to be smarter about dealing with transforms, buying efficiency?